python/lsst/ip/isr/isrFunctions.py

```diff
@@ -1222,3 +1224,65 @@ def countMaskedPixels(maskedIm, maskPlane):
     maskBit = maskedIm.mask.getPlaneBitMask(maskPlane)
     nPix = numpy.where(numpy.bitwise_and(maskedIm.mask.array, maskBit))[0].flatten().size
     return nPix
+
+
+def getExposureGains(exposure):
+    """Get the per-amplifier gains used for this exposure.
+
+    Parameters
+    ----------
+    exposure : `lsst.afw.image.Exposure`
+        The exposure to find gains for.
+
+    Returns
+    -------
+    gains : `dict` [`str` `float`]
+        Dictionary of gain values, keyed by amplifier name.
+    """
+    det = exposure.getDetector()
+    if det is None:
+        return dict()
+
+    metadata = exposure.getMetadata()
+    gains = {}
+    for amp in det:
+        ampName = amp.getName()
+        # The key may use the new LSST ISR or the old LSST prefix
+        if (key1 := f"LSST ISR GAIN {ampName}") in metadata:
+            gains[ampName] = metadata[key1]
+        elif (key2 := f"LSST GAIN {ampName}") in metadata:
```

Comment:
Both the old and new tasks have the same keys for "LSST ISR GAIN"

Reply:
Thanks. Isn't it possible that this code will run on data that has been processed with older versions of ISR where only one keyword is present?

Reply:
Eli suggests:
```I would leave in the backup check for backwards compatibility.  It’s the practical choice.```

Reply:
You and Eli are correct, keeping the original keys is fine.